executor/load_data.go

```diff
@@ -113,17 +121,160 @@ type LoadDataInfo struct {
 	IgnoreLines uint64
 	Ctx         sessionctx.Context
 	rows        [][]types.Datum
+
+	// these fields are used for pipeline data prepare and commit
+	commitTaskQueue chan CommitTask
+	QuitCommit      chan struct{}
```

Comment:
Could you just use a single quit channel ?

Reply:
Use two to let process and commit  both quit immediately when error occurs in one of them, commit will close `QuitProcess` and process will close `QuitCommit` .

Reply:
I think we can do it by only using one signal channel.